### PR TITLE
Local and Remote port range and csv #80

### LINKED ISSUE #80: remote_port and local_port should also support string for comma-separated port list or dash-separated port range.
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.5.x
- Ruby: 2.4.x
- Distribution: PE 2018.1
- Module version: 2.0.1

## How to reproduce (e.g Puppet code you use)
Assign local_port or remote_port a string containing comma-separated port number list (i.e. '80, 8080, 8081') or dash-separated port range ('8080-8089').
## What are you seeing
String type is not allowed for local_port and remote_port
## What behavior did you expect instead
Should be allowed
## Output log
parameter 'local_port' expects a value of type Undef, Integer[1, 65535], or Enum['any'], got String

### PR DESCRIPTION
**Pull request by sbezzy**
#### Pull Request (PR) description
Fix the ability to use a range of integers as well as comma separated integers and update the README to reflect those changes.

#### This Pull Request (PR) fixes the following issues

Fixes #80

### CODE REVIEW

**Comment by kenyon** [comment]
Looks good. What about adding some tests for this new functionality?

**Comment by sbezzy** [comment]
Thanks. I will work on that now.

File: manifests/exception.pp [hunk 1]
```diff
@@ -82,8 +82,8 @@
   Enum['allow', 'block'] $action = 'allow',
   Boolean $enabled = true,
   Optional[Enum['TCP', 'UDP', 'ICMPv4', 'ICMPv6']] $protocol = undef,
-  Optional[Variant[Stdlib::Port, Enum['any']]] $local_port = undef,
-  Optional[Variant[Stdlib::Port, Enum['any']]] $remote_port = undef,
+  Optional[Variant[Stdlib::Port, Enum['any'], Pattern[/\A[1-9]{1}\Z|[1-9]{1}[0-9,-]*[0-9]{1}\Z/]]] $local_port = undef,
```
Comment (alexjfisher) [inline manifests/exception.pp:85]: It look like it's worth creating a [type alias](https://puppet.com/docs/puppet/6.18/lang_type_aliases.html) for this.  eg `Windows_firewall::Port`.

You can even add a test for it. https://rspec-puppet.com/documentation/type_aliases/

Comment (sbezzy) [reply manifests/exception.pp:85]: Yeah, that's a good point @alexjfisher, it's pretty ugly. Working on it.

**Comment by kenyon** [comment]
replaced by #106

**Pull request closed without merging**